### PR TITLE
chore(cardinal): remove unused structs in CQL, define component getter function as type

### PR DESCRIPTION
**Pull request by technicallyty**
## Overview

defines the GetComponentByName dependency explicitly, so we dont type out the function signature multiple times

removes unused structs

### CODE REVIEW

**Review by pyrofolium — APPROVED** [review]
lgtm. Pretty straightforward.

**Review by namrapatel — APPROVED** [review]

**Comment by smsunarto** [comment]
I rebased this to main, update golangci to the new cql.go directory to continue ignoring it, and made one minor change:

Tyler's diff pointed the component getter was looking for "ComponentMetadata" when it should have been "Component". I fixed this accordingly.

File: cardinal/cql/cql.go [hunk 1]
```diff
@@ -248,12 +240,10 @@ func termToComponentFilter(
...
 
-func Parse(
-	cqlText string, stringToComponent func(string) (types.Component, error),
-) (filter.ComponentFilter, error) {
-	term, err := internalCQLParser.ParseString("", cqlText)
+func Parse(cqlText string, stringToComponent componentByName) (filter.ComponentFilter, error) {
+	term, err := internalCQLParser.ParseString("failed to parse CQL string", cqlText)
 	if err != nil {
-		return nil, eris.Wrap(err, "failed to parse CQL string")
+		return nil, eris.Wrap(err, "")
```
Comment (technicallyty) [inline cardinal/cql/cql.go:246]: ser @smsunarto 👁️ 👄 👁️

**Comment by technicallyty** [comment]
> I rebased this to main, update golangci to the new cql.go directory to continue ignoring it, and made one minor change:
> 
> Tyler's diff pointed the component getter was looking for "ComponentMetadata" when it should have been "Component". I fixed this accordingly. CQL doesn't need ComponentMetadata; Component is more than sufficient. ComponentMetadata creates tight coupling because it is intended for World internals.
> 
> If this is acceptable, should be good to merge the PR.

fine with me. Is there a plan to make metadata private? They both come from the same package, so am confused on the coupling part.

File: cardinal/cql/cql.go (hunk 1 above)
Comment (smsunarto) [reply cardinal/cql/cql.go]: whoops sorry

**Comment by technicallyty** [comment]
> > > I rebased this to main, update golangci to the new cql.go directory to continue ignoring it, and made one minor change:
> > > Tyler's diff pointed the component getter was looking for "ComponentMetadata" when it should have been "Component". I fixed this accordingly. CQL doesn't need ComponentMetadata; Component is more than sufficient. ComponentMetadata creates tight coupling because it is intended for World internals.
> > > If this is acceptable, should be good to merge the PR.
> > 
> > 
> > fine with me. Is there a plan to make ComponentMetadata private? They both come from the same package, so am confused on the coupling part.
> 
> ComponentMetadata is now moved to `cardinal/component` via my PR; this will create more clarity that this is an internal-facing type by moving it out of the user focused `cardinal` package.

ahhhh okay that makes sense. thx for clarifying🤠

**Pull request merged**